### ISSUE TITLE
what kind of menu

### ISSUE THREAD

**Issue by wdehoog** [sailfish-spotify/hutspot#11]
Currently hutspot has two kind of menus:

1. a docked panel with buttons, shown when pressing the 'burger' icon
2. a page with a list of items, shown when pressing the 'burger' icon or as an attached page of the player

Which one shall hutspot use? How will it be shown? Are there any other ways a menu could be shown ?

**Comment by rgrnetalk** [comment]
This was what confuses me most, but both option are nice (good to develop both of them!) 
1. the docked panel could come from under the header just like the filebrowser app does (that way is the distance between the button click and the actual panel much smaller). this way the panel can be used at every page without navigation confusion
2,  I used this option my adapted version and created a pullmenu choice menu which opens the menu. But is more confusing because the navigation is not that clear (there is no nice sequence in pages in the user scenario).

**Comment by rgrnetalk** [comment]
I have adapted the docker navigation panel to cover 2/3 of the page shown (the panel is opened by the first pulley menu choice. Here the result: 
![image](https://user-images.githubusercontent.com/12599622/44806180-89e28c00-abc6-11e8-8e74-9e9130f33a9c.png)
(the genre and mood page is indicated by the ambience icon and the devices are indicated by the devices icon).

**Comment by wdehoog** [comment]
Looks nice. Ambiance icon is a great idea. What is the problem with adding  a pulley to the pages that don't have one?

**Comment by rgrnetalk** [comment]
The LoadPullMenus has some other menu choices in it (Reload, Load Previous Set and Load Next Set) so that cannot be used for the devices and the settings page. Simply adding the pulleymenu to the page is not possible, because it has to open the component (id: navPanel) which it has no access to. Making a seperate pulleymenu component isn't possible because devices already has a menu (reload devices).

**Comment by wdehoog** [comment]
Devices and Settings are indeed sort of 'modal'. I guess if you want you could add 
```
    NavigationPanel {
        id: navPanel
    }
```
and 
```
        anchors.bottom: navPanel.top
        clip: navPanel.expanded
```

like the other pages have as well.

**Comment by wdehoog** [comment]
I have been working on a docked panel. It appears on most pages. When scrolling a list the panel will hide so more items are visible.

Currently it shows the album-art, prev/play/next buttons and a hamburger icon for showing the menu list.

I'd like to merge it but only if others like this change as well. 

It is in my own repo in the controlpanel2 branch. A build of my version can be found here: http://repo.merproject.org/obs/home:/wdehoog:/hutspot-wj/sailfish_latest_armv7hl/armv7hl/

![new-control-panel](https://user-images.githubusercontent.com/3353303/46833557-33fa2a00-cda9-11e8-8d39-c3fb1dab0cb5.jpg)

**Comment by ksiazkowicz** [comment]
I'm not a big fan of hamburger menus but honestly, looks pretty nice to me. There is a small bug with menu, it looks like you could swipe left to go to the next page, but there is no page. I guess we could merge it anyway and figure it out later.

**Comment by ksiazkowicz** [comment]
About pulley menus, it could be freed up with some kind of infinite loading mechanism.

**Comment by wdehoog** [comment]
merged. yes auto loading prev/next instead of pullies sounds nice.
I think the swipe left page indicator is the dimmed Accept one of the dialog. when showNavigationIndicator: false it is gone but then the Back one is also gone.

**Comment by ksiazkowicz** [comment]
Is scrolling behaving weird on your device as well? It looks like bottom pulley + docked panel results in a buggy mess:
https://1drv.ms/v/s!ApF6BBqdJeSZkLNlRRqb7-keP5p6VA

**Comment by wdehoog** [comment]
Yes. It is hard to have the docked panel sliding in and out while the push-up menus remain working and the last item not becoming inaccessible. I don't really know how to fix this.

**Comment by rgrnetalk** [comment]
My background panel has the following properties: 
height: navPanel.open? playingPage.height * 2/3 : col.height
opacity: navPanel.open ? 0.0 : 1.0
(but it's my adapted version with the menu panel sliding in from the bottom).
